DSA/Cpp/n_C_r_mod_P.cpp

```diff
@@ -0,0 +1,34 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+void solve()
+{ 
+    int n,r;
+    cin>>n>>r;
+    
+  const int mod=1e9+7;
+  int C[r+1];
+  memset(C,0,sizeof(C));
+  C[0]=1;
+  for(int i=0;i<=n;i++)
+  {
+      for(int j=min(i,r);j>0;j--)
+      {
+          C[j]=(C[j]+C[j-1])%mod;
+      }
+  }
+  if(!C[r])
+  cout<<C[r]<<endl;
+  else 
+  cout<<0<<endl;
+  return;
```

Comment:
correct your indentation and use variables like index, curr, inCurr, etc instead of i,j,k for loops.

Reply:
Ok sir I will make the required change, thank you for the review.